File: src/Services/MenuOrderService/NomNomGo.MenuOrderService.Domain/Entities/Order.cs
using System.ComponentModel.DataAnnotations;

namespace NomNomGo.MenuOrderService.Domain.Entities;

public class Order : BaseEntity
{
    [Required]
    public Guid UserId { get; set; }
        
    public Guid? CourierId { get; set; }
        
    [Required]
    public Guid RestaurantId { get; set; }
    public Restaurant Restaurant { get; set; } = null!;
        
    [Required]
    public decimal TotalPrice { get; set; }
        
    public decimal DiscountAmount { get; set; } = 0;
        
    [Required]
    public Guid StatusId { get; set; }
    public OrderStatus Status { get; set; } = null!;
        
    public DateTime? EstimatedDeliveryTime { get; set; }
        
    [MaxLength(500)]
    public string DeliveryAddress { get; set; } = string.Empty;
        
    [MaxLength(1000)]
    public string? Notes { get; set; }
        
    // Navigation
    public List<OrderItem> Items { get; set; } = new();
    public List<OrderStatusHistory> StatusHistory { get; set; } = new();
    public ActiveDelivery? ActiveDelivery { get; set; }
}

File: Clients/mobile courier/app/src/main/java/com/nomnomgo/courier/models/Order.kt
// Модель заказа согласно реальному ответу сервера
package com.nomnomgo.courier.models

import com.google.gson.annotations.SerializedName

data class CourierOrder(
    @SerializedName("id")
    val orderId: String?, // Guid от сервера

    @SerializedName("userId")
    val userId: String?,

    @SerializedName("courierId")
    val courierId: String?,

    @SerializedName("restaurant")
    val restaurant: RestaurantResponse?,

    @SerializedName("totalPrice")
    val totalPrice: Double = 0.0,

    @SerializedName("discountAmount")
    val discountAmount: Double = 0.0,

    @SerializedName("status")
    val status: String?,

    @SerializedName("estimatedDeliveryTime")
    val estimatedDeliveryTime: String?, // DateTime от сервера

    @SerializedName("deliveryAddress")
    val deliveryAddress: String?,

    @SerializedName("notes")
    val orderNotes: String?,

    @SerializedName("items")
    val items: List<OrderItemResponse> = emptyList(),

    @SerializedName("createdAt")
    val createdAt: String?
) {
    // Вспомогательные свойства для совместимости
    val restaurantName: String?
        get() = restaurant?.name

    val restaurantAddress: String?
        get() = restaurant?.address
}

data class RestaurantResponse(
    @SerializedName("id")
    val id: String,

    @SerializedName("name")
    val name: String,

    @SerializedName("address")
    val address: String,

    @SerializedName("phone")
    val phone: String?
)

data class OrderItemResponse(
    @SerializedName("id")
    val id: String,

    @SerializedName("name")
    val name: String,

    @SerializedName("quantity")
    val quantity: Int,

    @SerializedName("price")
    val price: Double
)